src/neuralNetwork/NeuralCore.ts

```diff
@@ -66,8 +67,42 @@ export class NeuralCore {
     this.createConnections(0, this.layerCnt - 1);
   }
 
-  public evaluate(input: number[]): number[] {
+  private resetWeights() {
+    let temp: number[][][];
+    this.weightList = temp;
+  }
+
+  public setWeights(weights: number[][][]) {
+    this.resetWeights();
+    if (weights.length !== this.layerCnt - 1) {
+      throw 'Weight count does not match layer count';
+    }
+    for (let i = 0; i < weights[0].length; i++) {
+      if (weights[0][i].length !== this.inputSize + 1) { //+1 for bias
+        throw `Weights at hidden layer 1 of neuron ${i+1} do not match the input count`;
+      }
+    }
+    for (let idx = 1; idx < this.hiddenLayerSizes.length; idx++) {
+      if (weights[idx].length !== this.hiddenLayerSizes[idx]) {
+        throw `Weights at hidden layer ${idx+1} do not match the neuron count`;
+      }
+      for (let i = 0; i < weights[idx].length; i++) {
+        if (weights[idx][i].length !== this.hiddenLayerSizes[idx-1] + 1) { //+1 for bias
+          throw `Weights at hidden layer ${idx+1} of neuron ${i+1} do not match the input count`;
+        }
+      }
+    }
+    for (let i = 0; i < weights[weights.length - 1].length; i++) {
+      if (weights[weights.length - 1][i].length !== this.hiddenLayerSizes[this.hiddenLayerSizes.length - 1] + 1) { //+1 for bias
+        throw `Weights at output layer of neuron ${i+1} do not match the input count`;
+      }
+    }
 
+    this.weightList = weights;
+  }
```

Comment:
This functions should loop through all the neurons and their connections and set the weights accordingly.